usaspending_api/awards/models.py

```diff
@@ -217,38 +219,66 @@ def get_or_create_summary_award(piid=None, fain=None, uri=None, awarding_agency=
                 # Now search for it
                 # Do we want to log something if the the query below turns up
                 # more than one award record?
-
                 if use_cache:
                     q_kwargs_fixed = list(q_kwargs.items()) + [('awarding_agency', awarding_agency), ]
                     q_kwargs_fixed.sort()
                     summary_award = awards_cache.get(q_kwargs_fixed)
                     if summary_award:
                         return [], summary_award
 
-                summary_award = Award.objects.all().filter(Q(**q_kwargs)).filter(awarding_agency=awarding_agency).first()
+                # Look for an existing award record
+                summary_award = Award.objects \
+                    .filter(Q(**q_kwargs)) \
+                    .filter(awarding_agency=awarding_agency) \
+                    .first()
+                if (summary_award is None and
+                        awarding_agency is not None and
+                        awarding_agency.toptier_agency.name != awarding_agency.subtier_agency.name):
+                    # No award match found when searching by award id info +
+                    # awarding subtier agency. Relax the awarding agency
+                    # critera to just the toptier agency instead of the subtier
+                    # agency and try the search again.
+                    awarding_agency_toptier = Agency.objects.filter(
```

Comment:
Yes, great idea! I have to revisit this for a late-breaking agency edge-case, so will take you up on this suggestion.